### ISSUE TITLE
`typeof C` type expression not supported

### ISSUE THREAD

**Issue by arv** [gajus/eslint-plugin-jsdoc#133]
Closure, TypeScript and Flow all support `typeof C` as a type expression.

```js
/**
 * @param {typeof C} constr
 * @return {C}
 */
function construct(constr) {
  return new constr();
}
```

**Comment by sandersn** [comment]
eslint-plugin-jsdoc relies on jsdoctypeparser to parse jsdoc types. I just opened [an umbrella issue for parsing all Typescript types there](https://github.com/Kuniwak/jsdoctypeparser/issues/50).

I'll link the PR I have in the works once it's ready. It adds `typeof` and `import("./package")` types.

**Comment by sandersn** [comment]
Fixed by #146
